agent/agentscripts/agentscripts.go

```diff
@@ -314,6 +320,15 @@ func (r *Runner) run(ctx context.Context, script codersdk.WorkspaceAgentScript)
 		} else {
 			logger.Info(ctx, fmt.Sprintf("%s script completed", logPath), slog.F("execution_time", execTime), slog.F("exit_code", exitCode))
 		}
+
+		_, err = r.scriptCompleted(ctx, &proto.WorkspaceAgentScriptCompletedRequest{
```

Comment:
It's possible this is `nil`, we should protect against this otherwise it'll panic.

Reply:
Which part of this could be `nil`?

Reply:
`r.scriptCompleted`

Reply:
Suggestion: In these cases I think it's nice to initialize with a noop function over handling nil explicitly where used. Not asking for a change though as I see this was already amended.